src/bullsandcows/Main.java

```diff
@@ -1,7 +1,194 @@
+        /*
+        We will generate ints and then convert them to valid characters from ascii table.
+
+        ASCII codes:
+
+        0-9: 48-57 (max possibleNumbers - 10)
+        a-z: 97-122 (max possibleNumbers - 36)
+        A-Z: 65-90 (max possibleNumbers - 62) - it makes game too complicated, but can be added at any moment
+
+         */
+        int initialNumbers = numbers;
+
+        Set<Character> answer = new LinkedHashSet<>();
+
+        if (numbers > possibleNumbers | numbers < 1 | possibleNumbers < 1)
+        {
+            System.out.printf("Can't generate a secret number with a length of %d with only %d unique symbols.%n", numbers, possibleNumbers);
+            System.out.println("Please enter a number of possible symbols no less than a number of symbols in secret.");
+            return answer;
+
+        } else
+        {
+
+            Random random = new Random(1);
+
+            while (numbers > 0)
+            {
+                char digit = encodeInt(random.nextInt(possibleNumbers));
+
+                // If answer already contains that digit - pass.
+                if (answer.contains(digit)) continue;
+
+                answer.add(digit);
+                numbers--;
+            }
+
+            String borderDigits = borderDigits(possibleNumbers);
+            String borderLetters = borderLetters(possibleNumbers);
+            System.out.printf("The secret is prepared: %s (%s%s).%n", new String(new char[initialNumbers]).replace("\0", "*"), borderDigits, borderLetters);
+            return answer;
+        }
+    }
+
+    public static char encodeInt(int code) {
+        char symbol = 0;
+        if (code < 10)
+        {
+            return String.valueOf(code).charAt(0);
+        } else
+        {
+            return (char) (code - 9 + 'a');
+        }
+    }
+
+    public static String borderDigits(int code) {
+        if (code < 2)
+        {
+            return code + "";
+        }
+
+        return 0 + "-" + Math.min(9, code-1);
+    }
+
+    public static String borderLetters(int code) {
+        if (code - 10 <= 0) return "";
+        if (code - 10 == 1) return ", a";
+
+        return ", a-" + (char) (code - 9 + 'a');
+    }
+
+    public static int grader(Set<Character> correctAnswer, String input){
+        int bulls = 0;
+        int cows = 0;
+        int size = correctAnswer.size();
+        Iterator<Character> answerIterator = correctAnswer.iterator();
+
+        for (int i = 0; i < size; i++) {
+            if (answerIterator.next() == input.charAt(i))
+            {
+                bulls += 1;
+            } else if (correctAnswer.contains(input.charAt(i)))
+            {
+                cows += 1;
+            }
+        }
+
+        System.out.printf("Grade: %d bull(s) and %d cow(s).%n", bulls, cows);
+
+        return bulls;
+    }
+
+    public static boolean isUserAnswerCorrect(String userAnswer, int numberLength) {
```

Comment:
is usually used for bean-like classes where is - an analog of a getter.
It is better to name this name like` checkUserAnswer(...)`